public/components/trace_analytics/components/services/services_content.tsx

```diff
@@ -61,6 +61,10 @@ export function ServicesContent(props: ServicesProps) {
   const [serviceTrends, setServiceTrends] = useState<ServiceTrends>({});
   const searchBarRef = useRef<{ updateQuery: (newQuery: string) => void }>(null);
 
+  useEffect(() => {
+    refresh();
+  }, [startTime, endTime, props.dataSourceMDSId]);
+
```

Comment:
Can we please merge this useEffect with the one on line : `83` 
https://github.com/opensearch-project/dashboards-observability/pull/2257/files#diff-6c41c2b6f0e21b1b29ad6d3cf28e92d0c5fc6bad2fbead527862b0634246ced8R83-R107

Reply:
Yes added the new dependencies to the use effect
```
    startTime,
    endTime,
    props.dataSourceMDSId,
```
This triggers the refresh